# " Assignment 1 "Python Basic 

<h2>1. In the below elements which of them are values or an expression? eg:- values can be integer or string and expressions will be mathematical operators.

<h3><b>*</b></h3>  Expression as it is mathematical operator <br>
    
<h3><b>'hello'</b> </h3>  Value  as it is string<br>
    
<h3><b>-87.8</b> </h3> Value as it is integer(float)<br>
    
<h3><b>-</b></h3>  Expression as it is mathematical operator<br>
    
<h3><b>/ </b></h3> Expressionas it is mathematical operator<br>
    
<h3><b>+</b></h3> Expression as it is mathematical operator<br>
    
<h3><b>6</b></h3> Value as it is integer


<h2>2. What is the difference between string and variable?</h2>

<b>
    
    String means any value that can be aplhabet or sequence of aplhabet or infact numeric which written inside quotes '' or double quotes "" .

    varible means a storage box which contains  our   data   and this data can be anything .And if we give any name to our varible than it called varible name as we  all have some names our varible box also have.


In [1]:
# String means any value that can be aplhabet or sequence of aplhabet or infact numeric which written inside quotes '' or double quotes "" .
# varible means a storage box which contains  our   data   and this data can be anything .And if we give any name to our varible than it called varible name as we  all have some names our varible box also have.

# lets understand by a example

Name= "Malik Asif" #  Name (is varible name) is varible which contain string called "Pooja Das" 
Class="12"    #  Class (is varible name) is varible which contain string called "10" as string can be anything written inside quotes"" or ''

print(Name ," ", Class)


Malik Asif   12


<h2>3. Describe three different data types.</h2>

In [3]:
Present=True # Boolean datatype represent only two value  True(1) or False(0)
Class= 100 #integer datatype is a numeric whole no. value that can be  positive por negative but without decimal
Id =11+6j  #complex type made with real + imaginary no. (11+6j)

print(type(Present))
print(type(Class))
print(type(Id))

<class 'bool'>
<class 'int'>
<class 'complex'>


<h2>4. What is an expression made up of? What do all expressions do?</h2>
        
        Expression  is made up of different numbers, varibles, and opertors and its use PEDMAS  Rule.(Parenthesis Exponents Division Multiplication Addition Subtraction)
        
        like R=[(a/10)*((a+b)+(b-5)+a)]
        
        And all expressions evaluate into a single value i.e R in above expression

In [4]:
a,b=20,10
R=[(a/10)*((a+b)+(b-5)+a)]


print(a/10) #20/10=2
print(a+b)  #20+10=30
print(b-5) #10-5=5
print((a+b)+(b-5)+a) #30+5+20=55
print([(a/10)*((a+b)+(b-5)+a)])
print("*****************************************************")
print("Expreesion value =",R) 
type(R) # List as it is enclosed  with [] 

2.0
30
5
55
[110.0]
*****************************************************
Expreesion value = [110.0]


list

<h2>5. This assignment statements, like spam = 10. What is the difference between an expression and a statement?</h2>
        
         Expression  is made with value, varible and opertors which evaluated and make a single value  like in  above example
         R=[(a/10)*((a+b)+(b-5)+a)] where expression evaluate into single value tht is R (110)
         
         but Statement does not evaluate like R=110 as 110 is assign to R  but in above example expression  evaluate and give R value as 110
    

<h2>6. After running the following code, what does the variable bacon contain?<br>
bacon = 22<br>
bacon + 1</h2>


In [4]:
bacon = 22 # 22 assign to bacon
bacon + 1 #22+1=23

23

<h2>7. What should the values of the following two terms be?<br>
'spam' + 'spamspam'<br>
'spam' * 3</h2>


In [5]:
print('spam' + 'spamspam') # both perfrom concatenation and give same result
print('spam' * 3)

spamspamspam
spamspamspam


<h2>8. Why is eggs a valid variable name while 100 is invalid?</h2>
        
        Because giving varible name it have some rules which we have to follow like<br>
        1. Varible can only start with string or it can be follow by underscore(_) like _eggs
        2. can't start with number  but  it can be follow by string like eggs100 
        3. can't be a keyword  like while , for , is , True,print,else  or etc
        4. can't contains symbol like @,#,%,$ , but can contain _
        5. no space between varible for space purpose we can use _ like eggs_100

In [6]:
#valid varible name
eggs=10
_eggs=10
eggs_100=10
eggs100=10


In [7]:
# not valid name
eggs@10=10 # no symbols

SyntaxError: cannot assign to operator (Temp/ipykernel_9480/403764228.py, line 2)

In [ ]:
100=10 #no number

In [8]:
while=100 # keywords we can see color also if it black than only valid if it   dark green  means it keyword

SyntaxError: invalid syntax (Temp/ipykernel_9480/3749558879.py, line 1)

<h2>9. What three functions can be used to get the integer, floating-point number, or string version of a value?</h2>
        
        int() for integer
        float() for floating-point number
        str() for string

In [9]:
A="100"
print("********************************Integer***************************")
B=int(A)
print("Integer is ",B)
print(type(B))
print("********************************Floating-point number***************************")
A=float(B)
print("Floating-point number is ",A)
print(type(A))
print("********************************STRING***************************")
B=str(A)
print("String is ",B)
print(type(B))

********************************Integer***************************
Integer is  100
<class 'int'>
********************************Floating-point number***************************
Floating-point number is  100.0
<class 'float'>
********************************STRING***************************
String is  100.0
<class 'str'>


<h2>10. Why does this expression cause an error? How can you fix it?<br>
'I have eaten ' + 99 + ' burritos.'</h2>
        
        It cause error because we can't concatenate string with integer for that we need to typecaste integer into string by str()


In [10]:
print('I have eaten ' + str(99) + ' burritos')

I have eaten 99 burritos
